### DISCUSSION TITLE
Ability to catch and resend DOM messages which are grid scoped.

### DISCUSSION THREAD

**Discussion by b-giavotto** [6pac/SlickGrid#1022] [Q&A]
### Clear and concise description of the problem

Is there a way to handle all slickgrid messages in a centralized way by catching and killing them, and upon a specific event (a async callback or a specified interval of time for example) resend them ?

Basically i need to capture a specific event, saving all event informations, then issue an async call to a server method, and after got the response re-create the same DOM or slickgrid event and send it to the grid as if it was the original one.

### Suggested solution

What i've asked

### Validations

- [X] Follow our [Code of Conduct](https://github.com/6pac/SlickGrid/blob/master/CODE_OF_CONDUCT.md)
- [X] Read the [Wikis](https://github.com/6pac/SlickGrid/wiki).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

**Comment by 6pac** [comment]
You can use the SlickGrid ```trigger``` method to trigger SlickGrid events manually. The return code of some events can be used to cancel the entire action, other events don't have this ability. DOM events can usually be cancelled with ```event.preventDefault``` or similar, and can be called manually. With both SlickGrid and DOM events, you'd have to save the event values somewhere for later re-use.

**Comment by 6pac** [comment]
Noticed you posted on StackOverflow too. I don't think there will be a generalised solution for all events because there would be some severe side effect from some events, eg mouse movements and scrolling, being piped through a round trip to a server. Also if an event is called multiple times you'd possibly need a queue structure for the parameters. It's a lot of extra complexity.   
So as I said, you'd want to target specific events, although you could write one sgement of generalised code to reroute those events. We can add cancelability to some SlickGrid actions if that would help.

**Comment by b-giavotto** [comment]
> Noticed you posted on StackOverflow too. I don't think there will be a generalised solution for all events because there would be some severe side effect from some events, eg mouse movements and scrolling, being piped through a round trip to a server. Also if an event is called multiple times you'd possibly need a queue structure for the parameters. It's a lot of extra complexity. So as I said, you'd want to target specific events, although you could write one sgement of generalised code to reroute those events. We can add cancelability to some SlickGrid actions if that would help.
 
I'm aware of the issues you highlighted above. However, i just asked if does exists, in slickgrid a centralized point when one could handle slickgrid events. Improper usage of those events and side effects, is of course up to me.

In the past i've done something similar to browser dom events applyed to single DOM nodes, and worked fine.
However, because slickgrid is more complex than a simple dom node i hoped for a more centralized way to catch events.
this would be beneficial because it allowed a more abstract, (at higher level), grid oriented way to do this.
Never checked the source code, but maybe there is a place where the grid handle DOM crude events to higher abstract event (such for example validated event which does not have a direct counterpart on DOM)

**Comment by ghiscoding** [comment]
This is a question, not an issue, so I'm moving this to a Discussion. 

Also this question was asked twice by you on Stack Overflow with no answer most probably because it's not very clear. SlickGrid uses SlickEvent which are just class wrapper of JavaScript events and those are synchronous. You can stop the event by using regular `e.preventDefault` but that won't stop all other subscribers to receive the event as well. What you can do though is to recreate the SlickEvent and resend it. The SlickEventHandler allows you to subscribe to multiple events and also has an `unsubscribeAll()` which is easier than unsubscribing 1 by 1. 

You won't get much more information than what we already provided, if that doesn't work for you, well sorry but that is all there is. You also mentioned that you never took a look at the source code, well I think you should so that you can better understand how the project is running. The [`slick.core.ts`](https://github.com/6pac/SlickGrid/blob/master/src/slick.core.ts) is where you will find the SlickEvent, SlickEventData and SlickEventHandler classes that you can subscribe to.

**Reply by b-giavotto** [reply]
OK, you keep missing the point , with your crude reply. I asked if there is a kind of centralized way to handle slickgrid events. 
I think is clear. but maybe i can get more informations by simply browsing source code.
Best Regards.
